Fix bug: Error, Expected signal to be an instanceof AbortSignal

bernatfortet opened the pull request:
- update dependencies (which require a new build): `node-fetch`, `abortcontroller-polyfill`, `typescript`, `es-lint`
- fix new errors/warnings caused by upgrading dependencies.
- re-build airtable.browser

vinayman commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "airtable",
-  "version": "0.12.2",
+  "name": "bernatfortet.airtable",
```
Believe you will want to remove `bernatfortet` from this

Also is there any reason the version has been bumped to 0.12.5 instead of 0.12.3?

bernatfortet commented:
@vinayman sorry for the late action. I've addressed the comments above.

@HongdaJiang-at @NabeelFarooqui-at could we see to merge this?

I see lots of related issues popping up.

HongdaJiang-at commented:
@bernatfortet Thank you for the improvement and apology for the delayed response here! This change looks good to me!

HongdaJiang-at commented on src/query.ts:
```diff
@@ -26,7 +26,7 @@ type RecordCollectionCallback<TFields extends FieldSet> = (
     records?: Records<TFields>
 ) => void;
-type DoneCallback = (error: CallbackError) => void;
+type DoneCallback = (error: CallbackError, other?: any) => void;
```
Curious why do we need this change here?

bernatfortet replied:
This was throwing a ts error when used in a ts project.

HongdaJiang-at commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "airtable",
-  "version": "0.12.2",
```
Do you mind revert this change and rebuild `airtable.browser.js)`? since our release process will automatically bump this version number up :)

bernatfortet commented:
Hi @HongdaJiang-at I run the `prepare` command. 
I also run `test` and I'm getting many errors, so unfortunately I'm not sure if this is ready to go out.
